subsys/usb/class/audio/audio.c

```diff
@@ -664,6 +818,97 @@ static int handle_interface_req(struct usb_setup_packet *pSetup,
 	return 0;
 }
 
+/**
+ * @brief Handler for endpoint sampling frequency control requests.
+ *
+ * This function handles endpoint sampling frequency control request.
+ *
+ * @param audio_dev_data USB audio device data.
+ * @param pSetup	 Information about the executed request.
+ * @param len		 Size of the buffer.
+ * @param data		 Buffer containing the request result.
+ *
+ * @return 0 if succesfulf, negative errno otherwise.
+ */
+static int handle_sampling_freq_req(struct usb_audio_dev_data *audio_dev_data,
+				    struct usb_setup_packet *pSetup,
+				    int32_t *len, uint8_t **data, uint8_t ep,
+				    uint8_t device)
+{
+	uint8_t *frequency = audio_dev_data->frequency[device];
+
+	switch (pSetup->bRequest) {
+	case USB_AUDIO_SET_CUR:
+		memcpy(frequency, *data, 3);
+		break;
+	case USB_AUDIO_GET_CUR:
+		memcpy(*data, frequency, 3);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* Inform the app */
+	if (audio_dev_data->ops &&
+	    audio_dev_data->ops->sampling_freq_update_cb) {
+		if (pSetup->bRequest == USB_AUDIO_SET_CUR) {
+			enum usb_audio_direction dir;
+			uint32_t freq;
+			freq = frequency[0] << 0;
```

Comment:
please use `freq = sys_get_le24(frequency);`

Reply:
was not aware of that function, very useful!